apis/v1alpha1/httproute_types.go

```diff
@@ -165,32 +180,68 @@ type HTTPRouteMatch struct {
 	ExtensionRef *RouteMatchExtensionObjectReference `json:"extensionRef" protobuf:"bytes,5,opt,name=extensionRef"`
 }
 
-// HTTPRouteFilter defines a filter-like action to be applied to
-// requests.
-type HTTPRouteFilter struct {
-	// Headers related filters.
-	//
-	// Support: extended
+const (
+	// FilterHTTPReqeustHeaderTransformer can be used to add or remove an HTTP
+	// header from an HTTP request before it is sent to the upstream target.
+	// Support: core
 	// +optional
-	Headers *HTTPHeaderFilter `json:"headers" protobuf:"bytes,1,opt,name=headers"`
+	FilterHTTPReqeustHeaderTransformer = "HTTPRequestHeaderTransformer"
+	// TODO(hbagdi): add FilterHTTPRequestMirroring filter
+)
+
+// HTTPRouteFilter defines additional processing steps that must be completed
+// during the request or response lifecycle.
+// HTTPRouteFilters are meant as an extension point to express additional
+// processing that must be done in Gateway implementations. Some examples include
+// request or response modification, implementing authentication strategies,
+// rate-limiting, and traffic shaping.
+// API guarantee/conformance is defined based on the type of the filter.
+// +union
+type HTTPRouteFilter struct {
+	// Type identifies the filter to execute.
+	// Types are classified into two conformance-levels:
+	// - Core and extended: These filter types and their corresponding configuration
+	//   is defined in this package. All implementations must implement
+	//   the core filters. Implementers are encouraged to support extended filters.
+	//   Definitions for filter-specific configuration for these
+	//   filters is defined in this package.
+	// - Custom: These filters are defined and supported by specific vendors.
+	//   In future, filters showing convergence in behavior across multiple
+	//   implementations will be considered for inclusion in extended or core
+	//   conformance rings. Filter-specific configuration can be specified
+	//   using ExtensionRef or Config fields.
+	// Implementers are encouraged to define custom implementation
+	// types to extend the core API with implementation-specific behavior.
+	//
+	// +unionDiscriminator
```

Comment:
I don't think I've seen this before, what does it do?

Reply:
IIUC enerator support for this is tracked in
* https://github.com/kubernetes-sigs/controller-tools/pull/298
* https://github.com/kubernetes-sigs/controller-tools/issues/461

Reply:
@hbagdi I suggest we add the above issues to the godocs for tracking purposes.

Reply:
Added these in.